my-react-app/src/redux/game/reducer.js

```diff
@@ -0,0 +1,43 @@
+import calculateWinner from '../../app/screens/Game/utils';
+
+import { actions } from './actions';
+
+const initialState = {
+  history: [{ squares: Array(9).fill(null) }],
+  stepNumber: 0,
+  xIsNext: true
+};
+
+function reducer(state = initialState, action) {
+  switch (action.type) {
+    case actions.handleClick: {
+      const history = state.history.slice(0, state.stepNumber + 1);
+      const current = history[history.length - 1];
+      const squares = current.squares.slice();
+      if (calculateWinner(squares) || squares[action.pos]) {
+        return state;
+      }
+      squares[action.pos] = state.xIsNext ? 'X' : 'O';
+      return {
+        history: history.concat([
+          {
+            squares
+          }
+        ]),
+        stepNumber: history.length,
+        xIsNext: !state.xIsNext
+      };
+    }
+    case actions.jumpTo: {
+      return {
+        history: state.history,
```

Comment:
Use the spread operator in this cases where you want to keep certain parts of the state as before

Reply:
This isn´t fixed yet. You need to use the spread operator in order to `return` the state as it was before, only with the `stepNumber` and `xIsNext` changed.

Reply:
This isn´t fixed yet. You need to use the spread operator in order to return the state as it was before, only with the stepNumber and xIsNext changed.